### PR TITLE
New Code formatting CI

### PR DESCRIPTION
**Pull request by gusty1g**
## Description

A breakdown of #3246 into smaller pull request.

I moved code formatting checks - yapf, flake8, mypy to run in a new workflow under single OS (ubuntu) and single python version (3.9). The new workflow checks only for formatting.

### CODE REVIEW

File: .github/workflows/new_formatting.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    - name: Install all dependencies using micromamba
+      uses: mamba-org/provision-with-micromamba@main
+      with:
+        environment-file: env.yml
+        environment-name: deepchem
+        channels: conda-forge,defaults
+        extra-specs: python=${{ matrix.python-version }}
+
+#    - name: Install all dependencies using conda
```
Comment (rbharath) [inline .github/workflows/new_formatting.yml:62]: Do we want this commented out portion?

**Review by rbharath — COMMENTED** [review]
To confirm, the only change in this PR is that the install/formatting checks now only run on ubuntu/3.9? How many minutes do we expect this to save us on the run?

Is there any edge case where running on mac could catch something this new test wouldn't?

File: .github/workflows/new_formatting.yml (hunk 1 above)
Comment (gusty1g) [reply .github/workflows/new_formatting.yml:62]: removed it as it is not needed.

**Comment by gusty1g** [comment]
> To confirm, the only change in this PR is that the install/formatting checks now only run on ubuntu/3.9? 

Yes

> How many minutes do we expect this to save us on the run?

A single code formatting run takes ~10 mins. Earlier, we had 3 runs on a pull request, 3 runs on a push ~ 60 mins. Now, a single run on pull request, a single run on push. ~20 mins. We save about 40 mins.

> Is there any edge case where running on mac could catch something this new test wouldn't?

Not any since the checks are platform independent.
